Add GRPluginMaps for ENGM Construtions summer 2025

Atriusftw opened the pull request:
Activated based on schedule according to SUP.
Closes #595

**Note that these maps depend on the APW from GRPlugin 1.6**
We feel that the new version is in such a state that we can use it for the next AIRAC.

![image](https://github.com/user-attachments/assets/953b2da1-a335-4434-822d-43a2c71a370a)

![image](https://github.com/user-attachments/assets/f0dc7603-9756-404e-aaa7-71d342b509fe)

![image](https://github.com/user-attachments/assets/304e6d07-ecd8-4c2d-9250-08a31bd04b53)

![image](https://github.com/user-attachments/assets/c49b038b-8a97-4e87-96b8-3bf15f71a191)

Sanderli25 commented on Plugins/GRpluginMaps.txt:
```diff
@@ -3826,3 +3826,77 @@ LINE:N060.11.28.595:E011.07.00.094:N060.11.27.792:E011.06.59.728
+// ENGM - Construction work at Gardermoen, the summer of 2025
+//
+
+MAP:REMOTE APRON CLOSURE
```
Maybe remove "closure" as it is specified by folder name. This applies to all following maps.

Sanderli25 commented on Plugins/GRpluginMaps.txt:
```diff
@@ -3826,3 +3826,77 @@ LINE:N060.11.28.595:E011.07.00.094:N060.11.27.792:E011.06.59.728
+FOLDER:AREA CLOSURE
+COLOR:CLOSUREAREA
+ACTIVE:250505:250512:0:0000:0000
+COORDTYPE:AREATYPE:REGION
```
Change from `AREATYPE` to `APW` to prevent need of defining `RESTRICITON:` line. This applies to all following maps.

Atriusftw replied:
AreaType `APM` doesn't work for me in version 1.5. Will test in 1.6 with `APW` tonight.

Sanderli25 commented on Plugins/GRpluginMaps.txt:
```diff
@@ -3826,3 +3826,77 @@ LINE:N060.11.28.595:E011.07.00.094:N060.11.27.792:E011.06.59.728
+LINE:N060.11.40.252:E011.06.08.270:N060.11.54.969:E011.06.16.828
+LINE:N060.11.54.969:E011.06.16.828:N060.11.56.043:E011.06.09.248
+
+MAP:STANDS 13,49 CLOSURE
```
`STANDS 13 & 49` ?

Sanderli25 reviewed (changes requested):
Good work! Few comments and proposals in the code.

Atriusftw commented:
![image](https://github.com/user-attachments/assets/b6f02c59-3675-44ee-8ea0-ee021f93a8f5)

Adrian2k commented:
Is there a way to disable stands for specified time periods?

Atriusftw commented:
Note to self: Need to add invisible boxes with ESE format around the taxiway lines to trigger the safety net.

Sanderli25 commented on Plugins/GRpluginMaps.txt:
```diff
@@ -3902,6 +3903,7 @@ AIRPORT:ENGM
 FOLDER:AREA CLOSURE
 COLOR:CLOSUREAREA
 ACTIVE:250801:250930:0:0000:0000
+BLOCKS:ENGM:28,60,64,68,72,76,80
```
Stand 28 should not be part of closure area. Remove it from `BLOCKS` and fix coordinates to stop below stand 60. Also maybe change MAP name to match what is learned in wiki: `MAP:NORTH PIER WEST` perhaps?

Sanderli25 commented on Plugins/GRpluginMaps.txt:
```diff
@@ -3913,6 +3915,7 @@ AIRPORT:ENGM
 FOLDER:AREA CLOSURE
 COLOR:CLOSUREAREA
 ACTIVE:250801:250930:0:0000:0000
+BLOCKS:ENGM:36,61,65,69,73,77,79,81,85,87,89,93,95
```
Stand 36 should not be part of closure area. Remove it from `BLOCKS` and fix coordinates to stop below stand 61. Also maybe change MAP name to match what is learned in wiki: `MAP:NORTH PIER EAST perhaps?`

Sanderli25 commented on Plugins/GRpluginMaps.txt:
```diff
@@ -3935,11 +3939,12 @@ COORD:N060.11.34.782:E011.06.28.335
 COORD:N060.11.33.702:E011.06.24.086
 
-MAP:STANDS 20-24
+MAP:STANDS 20-26
```
Should be 18-24 according to AIP SUP (`"Asphalt work on stands 18, 20, 22 and
24. Stands consecutively closed."`)

Sanderli25 commented on Plugins/GRpluginMaps.txt:
```diff
@@ -3935,11 +3939,12 @@ COORD:N060.11.34.782:E011.06.28.335
 FOLDER:AREA CLOSURE
 COLOR:CLOSUREAREA
 ACTIVE:250609:250630:0:0000:0000
+BLOCKS:ENGM:20,22,24,26
```
Blocks-line reflects map name, which is wrong. Fix:
`BLOCKS:ENGM:18,20,22,24`

Sanderli25 reviewed (commented):
Good work. Few things.

Sanderli25 commented:
I Will review once home later today, and is able to test with EuroScope. Code looks good.

Atriusftw commented:
To do:
* Check closure on V and P.
* Verify `ACTIVE` schedules. Might have to add one line per date.

Atriusftw commented:
> To do:
> 
>     * Check closure on V and P.
> 
>     * Verify `ACTIVE` schedules. Might have to add one line per date.

Latest commits implements the following proposed schedules:

![image](https://github.com/user-attachments/assets/6ea93f2e-c796-4ade-beb8-38df5420e1da)

Atriusftw commented:
![image](https://github.com/user-attachments/assets/62fa4d6a-cf04-486e-90fd-1d4805c23560)

Atriusftw commented:
Should be fine to merge together with #617 now.

Sanderli25 reviewed (commented):
Few missing closed/degraded lines on TWY V, P, G, H, Y and Z

Atriusftw commented:
Updated TWY closed maps
![image](https://github.com/user-attachments/assets/f0767f4d-835f-463b-b74a-db6f098b50a3)

Updated TWY degraded maps
![image](https://github.com/user-attachments/assets/fc004f2b-9a8d-4a87-9995-ee6349667249)

Sanderli25 commented:
From image you sent: TWY H closed shall be closed between TWY M and N as well?
When this is fixed, no more comments from my side. Well done, @Atriusftw 

<img width="175" alt="image" src="https://github.com/user-attachments/assets/c2da0739-19d2-4afc-92ec-a8101b3bf154" />

Atriusftw commented:
> From image you sent: TWY H closed shall be closed between TWY M and N as well? When this is fixed, no more comments from my side. Well done, @Atriusftw
> <img alt="image" width="175" src="https://private-user-images.githubusercontent.com/89723211/442529868-c2da0739-19d2-4afc-92ec-a8101b3bf154.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDY5OTEzNTYsIm5iZiI6MTc0Njk5MTA1NiwicGF0aCI6Ii84OTcyMzIxMS80NDI1Mjk4NjgtYzJkYTA3MzktMTlkMi00YWZjLTkyZWMtYTgxMDFiM2JmMTU0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA1MTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNTExVDE5MTczNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJjOGE1OGQ4YTQ0Zjk2ODk2MjgzMmM0ZmM0M2QyOTMyZDljYmEwNjM1ODlmYzY3YTVmNTI2ZmVkYTFjNGVmM2EmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.hmyVkMF5wddy9bXqX_VJbZFDnZSQQw6YM10BEIUydVE">

Added in [c532834](https://github.com/Vatsim-Scandinavia/ENOR-Sectorfile/pull/599/commits/c532834c70168f03ee1d757d583809b1b8913b88)
![image](https://github.com/user-attachments/assets/06f6dc38-9b5c-42c0-b745-50ae472c5aec)

Adrian2k commented on Plugins/GRpluginMaps.txt:
```diff
@@ -3826,3 +3827,971 @@ LINE:N060.11.28.595:E011.07.00.094:N060.11.27.792:E011.06.59.728
+AIRPORT:ENGM
+FOLDER:RWY CLOSURE
+COLOR:CLOSUREAREA
+ACTIVE:250516:250630:0:2230:0355
```
Looking at published NOTAM, it seems start of 22z is more correct?
![image](https://github.com/user-attachments/assets/1897979d-5624-4a1d-aa4c-4d32b859940a)

Atriusftw replied:
I don't think we had the NOTAM available when starting the work on this, hence we had to interpret the information from SUP to something that would work. Can change to 22z instead.

Adrian2k commented on Plugins/GRpluginMaps.txt:
```diff
@@ -3826,3 +3827,971 @@ LINE:N060.11.28.595:E011.07.00.094:N060.11.27.792:E011.06.59.728
+
+MAP:RWY 01R/19L
+AIRPORT:ENGM
+FOLDER:RWY CLOSURE
```
There is no active lines here.
Should be possible to activate this based on runway selection?
![image](https://github.com/user-attachments/assets/55e761eb-7275-40e3-923b-3d20754273d2)

``ACTIVE:RWY:ARR:*:01R,19L:DEP:*:01R,19L`` <- This should work I think?

Atriusftw replied:
01R/19L is not part of the summer closure. The map is added for future use.

Atriusftw replied:
Not sure if i agree on drawing a map that implies that the runway is **closed** when it's just not in use.

Adrian2k commented on Plugins/GRpluginMaps.txt:
```diff
@@ -3826,3 +3827,971 @@ LINE:N060.11.28.595:E011.07.00.094:N060.11.27.792:E011.06.59.728
+AIRPORT:ENGM
+FOLDER:AREA CLOSURE
+COLOR:CLOSUREAREA
+ACTIVE:250801:250930:0:0000:0000
```
"Stands consecutively closed."
So maybe better to seperate it into blocks.

Atriusftw replied:
Need a suggestion on the splits, then (relevant for all your comments on this)

Adrian2k replied:
Just to thow a suggestion out:

22-27 May - Stand 60,64,68

28May - 2 June - Stand 72, 76, 80

3-8 June - Stand 61, 65, 69

9-15 June - Stand 73, 77, 79, 81

16-22 June - Stand 85, 87, 89


Other variants can be done too, but with this we only need to create 5 polygons instead of 14+

Adrian2k replied:
Or with 4 polygons:

22.5 -> 29.5 - Stand 60 to 72
30.5 -> 6.6 - Stand 76 and 80
7.6 -> 14.6 - Stand 61 to 73
15.6 -> 22.6 - Stand 77-89

Atriusftw replied:
> Or with 4 polygons:
> 
> 22.5 -> 29.5 - Stand 60 to 72 30.5 -> 6.6 - Stand 76 and 80 7.6 -> 14.6 - Stand 61 to 73 15.6 -> 22.6 - Stand 77 to 89

Will opt for this suggestion for simplicity.

Atriusftw replied:
The dates you suggested does not match with the SUP, but i've split it now into the following:

**NORTH PIER WEST PT.1**
01.08 - 15.08: 60,64,68,72
**NORTH PIER WEST PT.2**
16.08 - 30.08: 76,80,96
**NORTH PIER EAST PT.1**
31.08 - 14.09: 61,65,69,73
**NORTH PIER EAST PT.2**
15.09 - 30.09: 77,79,81,85,87,89,93,95,96

Adrian2k commented on Plugins/GRpluginMaps.txt:
```diff
@@ -3826,3 +3827,971 @@ LINE:N060.11.28.595:E011.07.00.094:N060.11.27.792:E011.06.59.728
+AIRPORT:ENGM
+FOLDER:AREA CLOSURE
+COLOR:CLOSUREAREA
+ACTIVE:250801:250930:0:0000:0000
```
"Stands consecutively closed."
So maybe better to seperate it into blocks.

Adrian2k commented on Plugins/GRpluginMaps.txt:
```diff
@@ -3826,3 +3827,971 @@ LINE:N060.11.28.595:E011.07.00.094:N060.11.27.792:E011.06.59.728
+FOLDER:AREA CLOSURE
+COLOR:CLOSUREAREA
+ACTIVE:250609:250630:0:0000:0000
+BLOCKS:ENGM:18,20,22,24
```
"Stands consecutively closed."

but since its only 4 stands, I am good with just closing all 4 the whole period too.
